crates/binstalk-downloader/src/remote.rs

```diff
@@ -161,22 +204,33 @@ impl Client {
             .map_err(|err| Error::Http(Box::new(HttpError { method, url, err })))
     }
 
-    /// Check if remote exists using `method`.
-    pub async fn remote_exists(&self, url: Url, method: Method) -> Result<bool, Error> {
-        Ok(self
-            .send_request(method, url, false)
-            .await?
-            .status()
-            .is_success())
+    async fn head_or_fallback_to_get(&self, url: Url) -> Result<Response, Error> {
+        let res = self.send_request(Method::HEAD, url.clone(), true).await;
+
+        match res {
+            Err(Error::Http(http_error))
+                if http_error.err.status() == Some(StatusCode::METHOD_NOT_ALLOWED) =>
+            {
+                // Retry using GET
+                self.send_request(Method::GET, url, true).await
+            }
+            res => res,
+        }
     }
 
-    /// Attempt to get final redirected url.
+    /// Check if remote exists using `Method::HEAD` or `Method::GET` as fallback.
+    pub async fn remote_gettable(&self, url: Url) -> Result<bool, Error> {
+        self.head_or_fallback_to_get(url)
```

Comment:
Using `head_or_fallback_to_get` would help improve the perf and reduce http timeout/rate limit since we would only fallback if `GET` is not allowed (for amazon s3).